src/Microsoft.Health.Api/Features/Audit/AuditEventTypeMapping.cs

```diff
@@ -74,9 +74,20 @@ public Task StartAsync(CancellationToken cancellationToken)
                     return (ad.ControllerName, ad.ActionName, Attribute: attribute);
                 })
                 .Where(item => item.Attribute != null)
+                .Distinct()
+                .GroupBy(item => (item.ControllerName, item.ActionName))
+                .Select(item =>
```

Comment:
I think you can then merge this into the `GroupBy` and modify the group for easier use below, like:

```csharp
.GroupBy(
    x => (x.ControllerName, x.ActionName),
    x => x.Attribute,
    (key, values) =>
    {
        List<Attribute> attributes = values.ToList();
        if (attributes.Count > 1)
        {
            throw new DuplicateActionForAuditEventException(item.Key.ControllerName, item.Key.ActionName);
        }

        return (key, attributes[0]);
    }

Reply:
Thanks! Updated